utils/hare-fetch-fids

```diff
@@ -0,0 +1,164 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+# :help: Fetch the fids for services configured on a node
+
+import sys
+import re
+from typing import List, NamedTuple
+import json
+import argparse
+
+
+class Node:
+    def __init__(self, name: str):
+        self.name = name
+        self.svcs = []
+
+    def add_service(self, svc: str, fid: str):
+        self.svcs.append(Process(svc, fid))
+
+    def get_service(self, service: str):
+        for svc in self.svcs:
+            if svc.name == service:
+                return svc.fid
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(name={self.name}, svcs={self.svcs})'
+
+
+Process = NamedTuple('Process', [('name', str), ('fid', str)])
+
+
+def to_fid(container: int, key: int) -> str:
+    return f'{container:#x}:{key:#x}'
+
+
+def node_names(data: List) -> List[Node]:
+    # Get entries from file
+    # ​"key": "m0conf/nodes/<node_fid>",
+    # "value": "{​​​​​\"name\": \"<node name>\", \"state\": \"<HA state>\"}​​​​​"
+    regex = re.compile(
+        f'm0conf/nodes/[a-zA-Z0-9:]+$')
+    nodes = []
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        if len(key['key'].split('/')) == 3:
+            nodes.append(json.loads(key['value'])['name'])
+    return nodes
+
+
+def get_keys_for_processes(data: List) -> List[str]:
+    # Get 'm0conf/nodes/<node_id>/processes/<process_fidk>/services/<svc_type>' entries
+    # from the file.  See 'Consul KV Schema' in [4/KV](rfc/4/README.md).
+    regex = re.compile(
+        f'^m0conf\\/.*\\/processes\\/.*\\/services\\/(.+)$')
+    keys = []
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        keys.append(match_result.group(0))
+    return keys
+
+
+def add_service_to_node(nodes: List[Node], node_name: str, svc_name: str, fid: str):
+    for node in nodes:
+        if node.name == node_name:
+            node.add_service(svc_name, fid)
+
+
+def get_service_for_node(nodes: List[Node], node_name: str, service: str) -> str:
+    for node in nodes:
+        if node.name == node_name:
+            return node.get_service(service)
+
+
+def process_all_keys(keys: List, nodes: List[Node]):
+    names = {
+        'confd': 'confd',
+        'ios': 'ioservice',
+        'm0_client_s3': 's3server'
+    }
+    for key in keys:
+        # key looks like
+        # 'm0conf/nodes/<name>/processes/<process_fidk>/services/<svc_type>'
+        parts = key.split('/')
+        svc_type = parts[-1]
+        if svc_type in names:
```

Comment:
If key changes and the service type does not match the list we need to catch it instead of exiting silently. Lets log an exception in this case.

Reply:
Done, improved regex, and raising exceptions at appropriate places.